src/middlewared/middlewared/plugins/cache.py

```diff
@@ -223,7 +224,8 @@ async def refresh(self):
         for ds in available_ds:
             ds_state = await self.middleware.call(f'{ds}.get_state')
             if ds_state == 'HEALTHY':
-                await self.middleware.call(f'{ds}.fill_cache', True)
+                fill_job = await self.middleware.call(f'{ds}.fill_cache', True)
+                await fill_job.wait()
```

Comment:
If it errors out, would we silently ignore ? Should we perhaps use `wrap` here ?